cmd/kubeapps-apis/plugins/pkg/pkgutils/pkgutils.go

```diff
@@ -343,3 +345,28 @@ func isNonNullableNull(x interface{}, s *structuralschema.Structural) bool {
 func isKindInt(src interface{}) bool {
 	return src != nil && reflect.TypeOf(src).Kind() == reflect.Int
 }
+
+// translation to duration
+func ToDuration(duration string) (*metav1.Duration, error) {
+	if duration == "" {
+		return nil, nil
+	} else {
+		if d, err := time.ParseDuration(duration); err != nil {
+			return nil, err
```

Comment:
Yes, I agree it differs from what the Flux/Carvel CRs are doing, but for the core objects, it is in fact using duration expressed in seconds (see `gracePeriodSeconds`, `timeoutSeconds` or `expirationSeconds`).
OpenAPI seems to align also to the ISO format and [XML Schema](https://www.w3.org/TR/xmlschema11-2/#duration), although in Kubernetes they say: 

 ````
duration: a duration string like \"22 ns\" as parsed by Golang time.ParseDuration or compatible with Scala duration format`
````

That said, I don't think the UI should _display_ periods like `PT1H30M` but something more user-friendly. However, I do think the data sent over the wire should be using a standard instead of just using date parsing function that is only available in Golang. This way we would be forcing each client to build a Golang-alike duration parser, instead of just using the state-of-the-art existing date-handling dependency in the language the client uses.

If not... perhaps we should continue with the duration in seconds, so that the format is consistent across languages. 
Anyway, no strong opinion here though. Let's see what others think. cc @absoludity @gfichtenholt